src/movies.js

```diff
@@ -1,25 +1,89 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+
+function getAllDirectors(moviesArray) {
+
+ const getDir = moviesArray.map((el) => el.director)
+
+ return getDir
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+
+function howManyMovies(moviesArray) {
+    
+    const dramaSteve = moviesArray.filter((el) => el.director === "Steven Spielberg" && el.genre.includes("Drama") === true)
+
+    return dramaSteve.length
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+   if (!moviesArray.length) return 0; //return 0 if an empty array is passed
+
+   const avarScore = moviesArray.reduce((acc, val)=>{
+        if (!val.score)
+        {return acc} //return average even if one of the movies does not have score
+        else {return acc + val.score}
+   }, 0);
+
+   let avarScoreRounded = Math.round((avarScore/moviesArray.length)* 100)/ 100;
+   return avarScoreRounded;
+}
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+    
+    const dramaMovies = moviesArray.filter (movie => movie.genre.includes("Drama")) //1º Do an Array with JUST drama movies
+    return scoresAverage(dramaMovies)                                              //Return the average Scores of the array created
```

Comment:
Nice job returning the scoresAverage function inside of the dramaMoviesScore function!